### PR TITLE
Subscribe to Colony Controls

### PR DESCRIPTION
**Pull request by rdig**
## Description

This PR adds back the controls and logic to subscribe and unsubscribe from a colony. Along these, it also adds a colony information menu that is accessible when a user is subscribed to the colony

**Changes**

- [x] Added `ColonySubscription` subcomponent _(This contains the logic from the old `ColonyMeta`/`ColonySubscribe` components)_
- [x] Added `ColonySubscriptionPopover` subcomponent
- [x] Refactored core `MaskedAddress` in order to also display the full address

**Reviewer Note**

Rebased on `feature/dev-1` from #2280 

**Demo**

cs-colony-subscription

Resolves DEV-20

### CODE REVIEW

**Review by ceolson01 — COMMENTED** [review]
Just some a11y stuff again!

File: src/modules/core/components/MaskedAddress/MaskedAddress.tsx [hunk 1]
```diff
@@ -24,16 +24,32 @@ interface Props {
...
 
-const MaskedAddress = ({ address, mask = '...' }: Props) => {
+const MaskedAddress = ({ address, mask = '...', full = false }: Props) => {
   const cutAddress: AddressElements | Error = splitAddress(address);
   if (cutAddress instanceof Error) {
     return <FormattedMessage {...MSG.wrongAddressFormat} />;
   }
+  if (!full) {
+    return (
+      <span className={styles.address}>
```
[the remark this answers is not in the document]
Comment (rdig) [reply src/modules/core/components/MaskedAddress/MaskedAddress.tsx:42]: Sure

File: src/modules/dashboard/components/ColonyHome/ColonySubscription/ColonySubscription.tsx [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+          <div className={styles.spinnerContainer}>
+            <SpinnerLoader appearance={{ theme: 'primary', size: 'small' }} />
+          </div>
+        ))}
+      {!isSubscribed && (
+        <button
+          type="button"
+          className={styles.joinColony}
+          onClick={() => subscribe()}
+        >
```
[the remark this answers is not in the document]
Comment (rdig) [reply src/modules/dashboard/components/ColonyHome/ColonySubscription/ColonySubscription.tsx:83]: Yes, you are right here. I did it for the funds display (which was done after this), and meant to come back and fix it. Guess I forgot :(

File: src/modules/dashboard/components/ColonyHome/ColonySubscription/ColonySubscription.tsx [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+      )}
+      {isSubscribed && (
+        <ColonySubscriptionInfoPopover
+          colonyAddress={colonyAddress}
+          colonyDisplayName={colonyDisplayName || colonyName}
+          colonyName={colonyName}
+          nativeTokenAddress={nativeTokenAddress}
+          onUnsubscribe={() => unsubscribe()}
+        >
+          <div className={styles.menuIconContainer}>
```
[the remark this answers is not in the document]
Comment (rdig) [reply src/modules/dashboard/components/ColonyHome/ColonySubscription/ColonySubscription.tsx:95]: See other issues

**Review by ceolson01 — APPROVED** [review]
I meant to approve this last time - that's my bad!

**Pull request merged**